src/Http/GraphRequest.php

```diff
@@ -319,7 +325,7 @@ public function execute($client = null)
                 [
                     'body' => $this->requestBody,
                     'timeout' => $this->timeout
-                ]
+                ] + $this->getGuzzleSettings()
```

Comment:
On second thought, doing this would overwrite the configs(headers, http_errors etc) on a custom client passed in at a request level via `execute()`.

We could add a step to merge request config options only if `$this->guzzleClient` is not null.

v1 was designed to support guzzle client customisation at a request level which I agree is not the best experience. v2 (should be stable soon) allows you to customise the HTTP client when initialising the Graph client object.